lib/spack/spack/solver/asp.py

```diff
@@ -3807,6 +4022,11 @@ def build_specs(self, function_tuples):
         for s in self._specs.values():
             _develop_specs_from_env(s, ev.active_environment())
 
+        if spack.config.CONFIG.get("concretizer:splice:automatic"):
```

Comment:
Does this need to be guarded by re-checking the config, or can we key it on the concretizer output somehow? I'm (very mildly) concerned about whether the config could be changed in another process when someone is running multiple Spack processes in parallel and changing the config.

Reply:
It could be inferred from concretizer output. If there are no splice_at_hash facts, the rebuilder is a no-op anyway. So even if splicing is turned on, we could safely skip it.

Reply:
Or even if it's turned off we could safely make the call anyway?